indra/llcharacter/llik.cpp

```diff
@@ -1299,20 +1313,24 @@ void LLIK::Joint::shiftPos(const LLVector3& shift)
 
 void LLIK::Joint::setConfig(const Config& config)
 {
+    // we only remember the config here
+    // it gets applied later when we build the chains
     mConfig = &config;
     mConfigFlags = mConfig ? mConfig->getFlags() : 0;
 }
 
 void LLIK::Joint::resetFlags()
 {
     mConfig = nullptr;
-    // root Joint always has FLAG_LOCAL_ROT bit set
-    mConfigFlags = mParent ? 0 : FLAG_LOCAL_ROT;
+    mConfigFlags = 0;
+    // root Joint always has IK_FLAG_LOCAL_ROT_LOCKED set
+    mIkFlags = mParent ? 0 : IK_FLAG_LOCAL_ROT_LOCKED;
 }
 
 void LLIK::Joint::lockLocalRot(const LLQuaternion& local_rot)
 {
     mLocalRot = local_rot;
+    mIkFlags |= IK_FLAG_LOCAL_ROT_LOCKED;
     activate();
```

Comment:
I had previously stored this "local_rot_locked" bit in its own bool data member, then I moved it to another recycled flag.  Here I'm moving it out into its own bit in `mIkFlags` data member: it is almost the same as when an **LLIK::Joint::Confi**g has specified the local rotation of the **LLIK::Joint**... but is slightly distinct because we also want the root **Joint** to be locked during the IK calculations, even when it hasn't been configured for a specific local rotation.  This detail matters when we harvest the data and stream (through server) only the joint properties that have changed.